orttraining/orttraining/test/python/orttraining_test_ortmodule_execution_agents.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+from onnx import TensorProto
+from onnx import helper
+
+import torch
+
+import onnxruntime
+
+from onnxruntime.training.ortmodule._execution_agent import (
+    InferenceAgent,
+    TrainingAgent,
+)
+
+
+def make_inference_model():
+    inputs = [
+        helper.make_tensor_value_info("x", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias", TensorProto.FLOAT, [3]),
+    ]
+    mul_node = helper.make_node("Mul", ["x", "y"], ["mul"])
+    add_node = helper.make_node("Add", ["mul", "bias"], ["add"])
+    outputs = [helper.make_tensor_value_info("add", TensorProto.FLOAT, [2, 3])]
+    graph_def = helper.make_graph([mul_node, add_node], "graph", inputs, outputs)
+    return helper.make_model(graph_def)
+
+
+def make_training_model():
+    inputs = [
+        helper.make_tensor_value_info("x", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias", TensorProto.FLOAT, [3]),
+    ]
+    mul_node = helper.make_node("Mul", ["x", "y"], ["mul"])
+    add_node = helper.make_node("Add", ["mul", "bias"], ["add"])
+    yield_node = helper.make_node(
+        "YieldOp", ["add"], ["add_grad"], full_shape_outputs=[0], domain="com.microsoft"
+    )
+    x_grad_node = helper.make_node("Mul", ["add_grad", "y"], ["x_grad"])
+    y_grad_node = helper.make_node("Mul", ["x", "add_grad"], ["y_grad"])
+    axes_tensor = helper.make_tensor("axes", TensorProto.INT64, [1], [0])
+    bias_grad_node = helper.make_node(
+        "ReduceSum", ["add_grad", "axes"], ["bias_grad"], keepdims=0
+    )
+    outputs = [
+        helper.make_tensor_value_info("x_grad", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("y_grad", TensorProto.FLOAT, [2, 3]),
+        helper.make_tensor_value_info("bias_grad", TensorProto.FLOAT, [3]),
+    ]
+    graph_def = helper.make_graph(
+        [mul_node, add_node, yield_node, x_grad_node, y_grad_node, bias_grad_node],
+        "graph",
+        inputs,
+        outputs,
+        initializer=[axes_tensor],
+    )
+    return helper.make_model(graph_def)
+
+
+def assert_tensors_allclose(*args, **kwargs):
+    assert len(args) == 2
+    args = [t.detach().numpy() if isinstance(t, torch.Tensor) else t for t in args]
+    np.testing.assert_allclose(*args, **kwargs)
+
+
+def execution_agent_factory(cls):
```

Comment:
Try to keep only tests here. some versions of pytest really dont like to find functions that are not test in the same file

Reply:
Which version of pytest are we using? As far as I know, functions that doesn't start with `test_` are not considered tests. We can also configure pytest to look for more specific names
https://docs.pytest.org/en/6.2.x/example/pythoncollection.html

Reply:
It varies, in some pipelines it can be as old as 2.8.7

Reply:
How about marking them as `@pytest.fixture` then there should be no confusion?